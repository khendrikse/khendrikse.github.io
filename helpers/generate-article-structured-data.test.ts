import generateArticleStructuredData from './generate-article-structured-data';

describe('generateArticleStructuredData', () => {
  it('generates article structured data', () => {
    const input = {
      title: 'title',
      slug: 'fake-slug',
      description: 'this is a description',
      image: 'image.png',
      date: '2015-02-05T09:20:00+08:00'
    };

    const result = generateArticleStructuredData(input)
    expect(result).toMatchSnapshot();
  });
});
