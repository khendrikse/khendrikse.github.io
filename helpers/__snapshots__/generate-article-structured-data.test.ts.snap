// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateArticleStructuredData generates article structured data 1`] = `"{\\"@context\\":\\"https://schema.org\\",\\"@type\\":\\"BlogPosting\\",\\"mainEntityOfPage\\":{\\"@type\\":\\"WebPage\\",\\"@id\\":\\"https://velvety-tartufo-f1de54.netlify.app/fake-slug\\"},\\"headline\\":\\"title\\",\\"description\\":\\"this is a description\\",\\"image\\":\\"image.png\\",\\"datePublished\\":\\"2015-02-05T09:20:00+08:00\\",\\"dateModified\\":\\"2015-02-05T09:20:00+08:00\\",\\"author\\":{\\"@type\\":\\"Person\\",\\"name\\":\\"Karin Hendrikse\\",\\"url\\":\\"https://velvety-tartufo-f1de54.netlify.app/about\\"}}"`;
